### PR TITLE
Adding request body for listcredential for Microsoft.Kubernetes

### PR DESCRIPTION
**Pull request by akashkeshari**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2020-01-01-preview/connectedClusters.json [hunk 1]
```diff
@@ -607,6 +615,56 @@
...
+        "authenticationMethod",
+        "value"
+      ],
+      "properties": {
+        "authenticationMethod": {
+          "type": "string",
+          "description": "The mode of client authentication.",
+          "enum": [
+            "Token",
+            "Certificate"
```
Comment (raunakpandya) [inline specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2020-01-01-preview/connectedClusters.json:630]: >Certificate [](start = 13, length = 11)

ClientCertificate

File: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2020-01-01-preview/examples/ConnectedClustersListClusterCredentialsResult.json [hunk 2]
```diff
@@ -3,7 +3,16 @@
     "subscriptionId": "1bfbb5d0-917e-4346-9026-1d3b344417f5",
     "resourceGroupName": "k8sc-rg",
     "clusterName": "testCluster",
-    "api-version": "2020-01-01-preview"
+    "api-version": "2020-01-01-preview",
+    "ClientAuthenticationDetails": {
+      "authenticationMethod": "Certificate",
```
Comment (raunakpandya) [inline specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2020-01-01-preview/examples/ConnectedClustersListClusterCredentialsResult.json:8]: >Certificate [](start = 31, length = 11)

Can you add another example here and leave the default example with no body? No bodyworks for AAD scenario. For non AAD case user needs to pass in credential

**Review by raunakpandya — CHANGES_REQUESTED** [review]
:clock1:

File: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2020-01-01-preview/connectedClusters.json (hunk 1 above)
Comment (akashkeshari) [reply specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2020-01-01-preview/connectedClusters.json]: updated

File: specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2020-01-01-preview/examples/ConnectedClustersListClusterCredentialsResult.json (hunk 2 above)
Comment (akashkeshari) [reply specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2020-01-01-preview/examples/ConnectedClustersListClusterCredentialsResult.json]: added

**Review by raunakpandya — APPROVED** [review]
:shipit:

**Comment by chiragg4u** [comment]
Please go through the contribution checklist added as part of the review and mark them completed. This is a requirement before review can be continued.

**Comment by akashkeshari** [comment]
> Please go through the contribution checklist added as part of the review and mark them completed. This is a requirement before review can be continued.

Will require you approval on the PR in order to request the SDK release. https://portal.azure-devex-tools.com/tools/request-api-release

Please approve

**Pull request merged**